### PR TITLE
EZP-29143 run test features on Demo version without changing feature files

### PR DESCRIPTION
**Pull request by m-tyrala**
Adding `PlatformDemoEnvironmentConstants` class.
JIRA: https://jira.ez.no/browse/EZP-29143

### CODE REVIEW

File: tests/AppBundle/Behat/PlatformDemoEnvironmentConstants.php [hunk 1]
```diff
@@ -0,0 +1,26 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace Tests\AppBundle\Behat;
+
+use EzSystems\EzPlatformAdminUi\Behat\Helper\InstallType;
```
Comment (mnocon) [inline tests/AppBundle/Behat/PlatformDemoEnvironmentConstants.php]: Can be removed?

File: tests/AppBundle/Behat/PlatformDemoEnvironmentConstants.php [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+class PlatformDemoEnvironmentConstants
```
Comment (mnocon) [inline tests/AppBundle/Behat/PlatformDemoEnvironmentConstants.php]: Just an idea: what about inheriting from PlatformEnvConsts, invoking parent constructor and then overriding only the variables that differ? We should do something similar in PageBuilder too.

Advantage: We would avoid defining the same variables (with same values) in many places
Disadvantage: there is a dependency on AdminUI in metarepo - that's ok for me, but needs @damianz5 approval

File: composer.json [hunk 3]
```diff
@@ -35,17 +35,17 @@
...
+        "ezsystems/repository-forms": "~2.2.0@dev",
+        "ezsystems/ezplatform-solr-search-engine": "~1.5.0@dev",
+        "ezsystems/ez-support-tools": "~0.2.1@dev",
+        "ezsystems/ezplatform-http-cache": "~0.7.0@dev",
+        "ezsystems/ezplatform-admin-ui": "~1.2.0@dev",
+        "ezsystems/ezplatform-admin-ui-modules": "~1.2.0@dev",
+        "ezsystems/ezplatform-admin-ui-assets": "~3.0.0@dev",
+        "ezsystems/ezplatform-design-engine": "~2.0.0@dev",
+        "ezsystems/ezplatform-standard-design": "~0.1.0@dev",
+        "ezsystems/ezplatform-cron": "~2.0.0@dev",
```
Comment (mnocon) [inline composer.json]: I think this changes should come when ezplatform is merged to ezplatform-demo (should not be set directly here).

File: .env [hunk 4]
```diff
@@ -4,7 +4,7 @@ COMPOSE_DIR=.
 
 # You'll need to adjust this for Windows and XDB Linux systems: https://getcomposer.org/doc/03-cli.md#composer-home
 COMPOSER_HOME=~/.composer
-COMPOSER_MEMORY_LIMIT=2G
+COMPOSER_MEMORY_LIMIT=8g
```
Comment (andrerom) [inline .env]: Hope we don't need that much :)

**Comment by m-tyrala** [comment]
I consider tests conditionally passed - there's only one unrelated error in the temporary build for testing it along with the other PRs [here](https://travis-ci.org/ezsystems/ezplatform-demo/builds/419058436). Problem is already reported in [EZP-29549](https://jira.ez.no/browse/EZP-29549). I'll clear this PR from this temporary changes now.

**Comment by m-tyrala** [comment]
@lserwatka this one can be merged too

File: bin/.travis/trusty/setup_ezplatform.sh [hunk 5]
```diff
@@ -91,7 +91,7 @@ if [[ -n "${DEPENDENCY_PACKAGE_NAME}" ]]; then
 fi
 
 echo '> Run composer install inside docker app container'
-docker-compose exec app sh -c 'composer install --no-suggest --no-progress --no-interaction --prefer-dist --optimize-autoloader'
+docker-compose exec app sh -c 'php -d memory_limit='3G' /usr/local/bin/composer install --no-suggest --no-progress --no-interaction --prefer-dist --optimize-autoloader'
```
Comment (andrerom) [inline bin/.travis/trusty/setup_ezplatform.sh]: Isn't this change unneeded as we now use env variable for this?
https://github.com/ezsystems/ezplatform/commit/add8d279f1d7b3b2666bb72935ff23c795bf812c

Comment (m-tyrala) [reply bin/.travis/trusty/setup_ezplatform.sh]: Sadly this env variable does not take effect here for some reason. We had the error with the lack of memory with the default composer memory limit value in the message until I added this option here.

**Pull request merged**